python/pyspark/tests/test_worker.py

```diff
@@ -144,6 +144,15 @@ def test_with_different_versions_of_python(self):
         finally:
             self.sc.pythonVer = version
 
+    def test_reuse_worker_of_parallelize_xrange(self):
```

Comment:
Can you please make a separate test for such tests?

Reply:
I mean, separate class. It's best to avoid reusing Spark context.

Reply:
No problem, done in 4868e82.